### PR TITLE
Add support for Sega Master System and Game Gear

### PR DESCRIPTION
**Pull request by lidnariq**
Here's tested support for Sega's Master System and Game Gear VDP layouts

### CODE REVIEW

File: src/Map.h [hunk 1]
```diff
@@ -80,6 +80,13 @@ inline byte_vec_t pack_native_mapentry(const Mapentry& entry, Mode mode) {
     v.push_back(entry.tile_index & 0xff);
     break;
 
+  case Mode::sms:
+  case Mode::gg:
+    v.push_back(entry.tile_index & 0xff);
+    v.push_back(((entry.tile_index >> 8) & 0x01) | (entry.flip_h << 1) | (entry.flip_v << 2) | ((entry.palette_index << 3) & 0x8));
+    // SMS and GG support depth information per tile in tilemap, instead of per sprite. But superfamiconv doesn't provide that rope?
```
Comment (Optiroc) [inline src/Map.h:87]: You got me there!

Comment (Optiroc) [reply src/Map.h:87]: I will have to look into the specifics and see if that's something that can be supported.

**Review by Optiroc — APPROVED** [review]

File: src/Mode.h [hunk 2]
```diff
@@ -471,10 +507,15 @@ inline byte_vec_t pack_native_color(const rgba_t color, Mode mode) {
...
   case Mode::wsc:
+  case Mode::gg:
   case Mode::wsc_packed:
     v.push_back(((color >> 16) & 0x0F) | ((color >> 4) & 0xF0));
     v.push_back((color & 0x0F));
     break;
+  case Mode::sms:
+    v.push_back(((color >> 12) & 0x30) | ((color >> 6) & 0x0C) | (color & 3));
+    break;
+  case Mode::pv1k:
```
Comment (Optiroc) [inline src/Mode.h]: What's this mode? 🤔

**Review by Optiroc — CHANGES_REQUESTED** [review]

**Comment by Optiroc** [comment]
Thanks for the PR! Looking good, but there's one stray reference to `Mode::pv1k` that seems quite out of place.

File: src/Mode.h (hunk 2 above)
Comment (lidnariq) [reply src/Mode.h]: It's how you know I didn't re-build after cherry-picking the SMS support away from the PV-1000 support :sigh:

**Pull request merged**